sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/azure_configuration_client_async.py

```diff
@@ -324,18 +456,72 @@ def list_configuration_settings(
-        if_match = quote_etag(etag) if etag else None
-        key_value_deleted = await self._impl.delete_key_value(
-            key=key,
-            label=label,
-            if_match=if_match,
-            headers=custom_headers,
-            error_map={
-                409: ResourceReadOnlyError,
-                412: ResourceModifiedError,
-            },
-        )
-        return ConfigurationSetting._from_key_value(key_value_deleted)
+        if match_condition == MatchConditions.Unconditionally:
+            key_value_deleted = await self._impl.delete_key_value(
+                key=key,
+                label=label,
+                headers=custom_headers,
+                error_map={
+                    409: ResourceReadOnlyError,
+                },
+            )
+            return ConfigurationSetting._from_key_value(key_value_deleted)
+
+        if match_condition == MatchConditions.IfNotModified:
+            if_match = quote_etag(etag) if etag else None
+            key_value_deleted = await self._impl.delete_key_value(
+                key=key,
+                label=label,
+                if_match=if_match,
+                headers=custom_headers,
+                error_map={
+                    409: ResourceReadOnlyError,
+                    412: ResourceModifiedError,
+                },
+            )
+            return ConfigurationSetting._from_key_value(key_value_deleted)
+
+        if match_condition == MatchConditions.IfModified:
+            if_none_match = quote_etag(etag) if etag else None
+            key_value_deleted = await self._impl.delete_key_value(
+                key=key,
+                label=label,
+                if_none_match=if_none_match,
+                headers=custom_headers,
+                error_map={
+                    409: ResourceReadOnlyError,
+                    412: ResourceNotModifiedError,
+                },
+            )
+            return ConfigurationSetting._from_key_value(key_value_deleted)
+
+        if match_condition == MatchConditions.IfPresent:
+            if_match = "*"
+            key_value_deleted = await self._impl.delete_key_value(
+                key=key,
+                label=label,
+                if_match=if_match,
+                headers=custom_headers,
+                error_map={
+                    409: ResourceReadOnlyError,
+                    412: ResourceNotFoundError,
+                },
+            )
+            return ConfigurationSetting._from_key_value(key_value_deleted)
+
+        if match_condition == MatchConditions.IfMissing:
+            if_none_match = "*"
+            key_value_deleted = await self._impl.delete_key_value(
+                key=key,
+                label=label,
+                if_none_match=if_none_match,
+                headers=custom_headers,
+                error_map={
+                    409: ResourceReadOnlyError,
+                    412: ResourceExistsError,
+                },
+            )
```

Comment:
This kind of code is repeated both across sync/async but also repeated with both across operations. What about factoring out functions that construct kwargs?

    key_value_deleted = await self._impl.delete_key_value(
        key=key,
        label=label, 
        headers=custom_headers,
        **match_args(match_condition, etag))

Where `match_args` would return something like `{'if_match': ..., 'error_map': ...}` Such a function would also be easy to put under comprehensive unit tests. The only caveat is that the error maps would need to be the union of all error maps for a given match condition, but assuming passing irrelevant entries to operations where they don't apply is harmless.